book/python/writing-your-own-stateful-application.md

```diff
@@ -1,96 +1,70 @@
+### State
 
-1. Your state computation might not have updated the state, in which case saving its state for recovery is wasteful.
-2. You might only want to save after some changes. Saving your state can be expensive for large objects. There's a trade-off that can be made between performance and safety.
-
-### State and StateBuilder
-
-The state for this application is two-tiered. There is a vote count for each letter:
+The state for this application keeps track of a running total of transactions:
 
 ```python
-class Votes(object):
-    def __init__(self, letter, votes):
-        self.letter = letter
-        self.votes = votes
+class TransactionTotal(object):
+    total = 0
 ```
 
-And a state map:
-
-```python
-class AllVotes(object):
-    def __init__(self):
-        self.votes_by_letter = {}
-
-    def update(self, votes):
-        letter = votes.letter
-        vote_count = votes.votes
-
-        votes_for_letter = self.votes_by_letter.get(letter, Votes(letter, 0))
-        votes_for_letter.votes += vote_count
-        self.votes_by_letter[letter] = votes_for_letter
-
-    def get_votes(self, letter):
-        vbl = self.votes_by_letter[letter]
-        return Votes(letter, vbl.votes)
-```
-
-This map is the `state` object that `AddVotes.compute` above takes.
-An important thing to note here is that `get_votes` returns a _new_ `Votes` instance. This is important, as this is the value that is eventually passed to `Encoder.encode`, and if we passed a reference to a mutable object here, there is no guarantee that `Encoder.encode` will execute before another update to this object.
-
 ### Encoder
-The encoder is going to receive a `Votes` instance and encode into a string of the form `LETTER => VOTES\n` where `LETTER` is the letter and `VOTES` is the number of votes for said letter. Each incoming message generates one output message.
-As with the previous example, the sink requires a `bytes` object. In Python 2 this can be the string itself, but in Python3 we need to encode it from unicode to `bytes`. Luckily, we can use `encode()` to get a `bytes` from a string in both versions:
+The encoder is going to receive either a `DepositAlert` or `WithdrawalAlert` instance and encode it into a string. Since we only generate alerts when certain conditions are met, not every input into the application results in an output sent to teh sink.
+
+As with our previous stateless example, the sink requires a `bytes` object. In Python 2 this can be the string itself, but in Python3 we need to encode it from unicode to `bytes`. Luckily, we can use `encode()` to get a `bytes` from a string in both versions:
```

Comment:
I think by convention we've been using "Python 3" rather than "Python3".

Reply:
There are numerous other instances, but rough guess is they're not in the majority.  I went through and updated them.